File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int out = 0, l;
        unordered_set<char> charSet;
        for(int r = 0; r < s.size(); r++)
        {
            while ( charSet.find(s[r]) != charSet.end())
            {
                charSet.erase(s[l]);
                l++;
            }
            charSet.insert(s[r]);
            out = max(out, r - l + 1);
        }
        return out;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s)
    {
        if (s.Length == 0) return 0;
        int l = 0, r = 0, result = 0;
        string substr = "";
        for (;r < s.Length;)
        {
            while (substr.Contains(s[r]))
            {
                substr = substr.Remove(0, 1);
                l++;
            }
            substr += s[r];
            result = Math.Max(result, r - l + 1);
            r++;
        }
        return result;
    }
}